packages/algorand-js/src/interpreter/opcode-list.ts

```diff
@@ -545,13 +578,52 @@ export class Substring3 extends Op {
     const start = this.assertBigInt(stack.pop());
     const end = this.assertBigInt(stack.pop());
 
-    if (end < start) {
-      throw new TealError(ERRORS.TEAL.SUBSTRING_END_BEFORE_START);
-    }
-    if (start > byteString.length || end > byteString.length) {
-      throw new TealError(ERRORS.TEAL.SUBSTRING_RANGE_BEYOND);
-    }
+    const subString = this.subString(start, end, byteString);
+    stack.push(subString);
+  }
+}
+
+/** Pseudo-Ops **/
 
-    stack.push(byteString.slice(Number(start), Number(end)));
+// push integer to stack
+export class Int extends Op {
+  readonly uint64: bigint;
+
+  constructor (uint64: bigint) {
+    super();
+    this.uint64 = uint64;
+  }
+
+  execute (stack: TEALStack): void {
+    this.checkOverflow(this.uint64);
+    stack.push(this.uint64);
+  }
+}
+
+// push bytes to stack
+export class Byte extends Op {
```

Comment:
I think this has some different function, can you please confirm ?
```
txn Lease
byte base64 TMPL_LEASE
==
&&
```
https://developer.algorand.org/docs/reference/teal/templates/dynamic_fee/

Reply:
Yes. But these should be handled by the parser. There are a lot of variations for `byte`.
Check: [https://developer.algorand.org/docs/reference/teal/specification/#constants-and-pseudo-ops]()

This pseudo-op is taking parsed byte (as `Uint8array`) and pushing it to stack.

Reply:
Update: Updated pseudo ops to do conversion.